qiskit/visualization/circuit/_utils.py

```diff
@@ -506,32 +506,34 @@ def _sorted_nodes(dag_layer):
 
 
 def _get_gate_span(qubits, node, measure_arrows):
-    """Get the list of qubits drawing this gate would cover
-    qiskit-terra #2802
-    """
-    min_index = len(qubits)
-    max_index = 0
-    for qreg in node.qargs:
-        index = qubits.index(qreg)
-
-        if index < min_index:
-            min_index = index
-        if index > max_index:
-            max_index = index
+    """Return the ordered wires this node would occupy when drawn."""
 
     if isinstance(node.op, ControlFlowOp) and not isinstance(node.op, BoxOp):
         # Because of wrapping boxes for mpl control flow ops, this
         # type of op must be the only op in the layer
         # BoxOps are excepted because they have one block executed unconditionally
-        span = qubits
-    elif node.cargs and (
-        (measure_arrows and isinstance(node.op, Measure)) or getattr(node.op, "condition", None)
+        return qubits
+
+    qubit_index_map = {bit: idx for idx, bit in enumerate(qubits)}
+    qubit_indices = [qubit_index_map[qarg] for qarg in node.qargs if qarg in qubit_index_map]
```

Comment:
Computing the dict, and computing the indices off of this dict is also a nice performance improvement, thanks! My only comment is that I don't understand under which circumstances `if qarg in qubit_index_map` may be `False`, I tried your code without this `if` and it still seems to work great.

```suggestion
    qubit_index_map = {bit: idx for idx, bit in enumerate(qubits)}
    qubit_indices = [qubit_index_map[qarg] for qarg in node.qargs]
```